src/Parser/MarksmanshipHunter/Modules/FocusChart/Focus.js

```diff
@@ -0,0 +1,374 @@
+//Based on Main/Mana.js and Parser/VengeanceDemonHunter/Modules/PainChart
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import ChartistGraph from 'react-chartist';
+import Chartist from 'chartist';
+import 'chartist-plugin-legend';
+
+import makeWclUrl from 'common/makeWclUrl';
+import SPELLS from 'common/SPELLS';
+
+import specialEventIndicators from 'Main/Chartist/specialEventIndicators';
+
+import 'Main/Mana.css';
+
+import FocusComponent from './FocusComponent';
+import './Focus.css';
+const formatDuration = (duration) => {
+  const seconds = Math.floor(duration % 60);
+  return `${Math.floor(duration / 60)}:${seconds < 10 ? `0${seconds}` : seconds}`;
+};
+
+const passiveWasteThreshold = .03; // (wasted passive focus generated) / (total passive focus generated), anything higher will trigger "CAN BE IMPROVED"
+//TODO: get a "real" number approved by a MMS expert
+
+class Focus extends React.PureComponent {
+  static propTypes = {
+    reportCode: PropTypes.string.isRequired,
+    actorId: PropTypes.number.isRequired,
+    start: PropTypes.number.isRequired,
+    end: PropTypes.number.isRequired,
+  };
+  constructor() {
+    super();
+    this.state = {
+      focusData: null,
+      bossHealth: null,
+    };
+	
+  }
+  
+
+  componentWillMount() {
+    this.load(this.props.reportCode, this.props.actorId, this.props.start, this.props.end);
+  }
+  componentWillReceiveProps(newProps) {
+    if (newProps.reportCode !== this.props.reportCode || newProps.actorId !== this.props.actorId || newProps.start !== this.props.start || newProps.end !== this.props.end) {
+      this.load(newProps.reportCode, newProps.actorId, newProps.start, newProps.end);
+    }
+  }
+  load(reportCode, actorId, start, end) {
+    const focusPromise = fetch(makeWclUrl(`report/tables/resources/${reportCode}`, {
```

Comment:
This data should be available in the existing events handler. Please use that instead. You can for the time being leave this as it is (since I estimate changing it will take some time), but please do at some point change this as additional requests are precious.